t/01-test_needle.t

```diff
@@ -214,7 +228,22 @@ ok(!defined $res, "the headline is completely different");
 
 $img1   = tinycv::read($data_dir . "inst-rescuesystem-20141027.test.png");
 $needle = needle->new($data_dir . "inst-rescuesystem-20141027.json");
-$res    = $img1->search($needle);
+($res, $cand) = $img1->search($needle);
+is_deeply(
+    $cand->[0]->{area},
+    [
+        {
+            similarity => '0',
+            x          => 245,
+            w          => 312,
+            result     => 'fail',
+            y          => 219,
+            h          => 36
+
+        }
+    ],
+    'candidate total fail, but not at 0x0'
```

Comment:
Sorry, this check is just ...

The old code already returned 0,0 when the best match had an error > INT_MAX, i.e. an arbitrary value, depending on the needle size. Now requiring the method to always return the best match (which hopefully is better than error=INT_MAX), and the best match to be something different to 0,0 is completely bogus.